slack-message.c

```diff
@@ -441,14 +445,17 @@ void slack_handle_message(SlackAccount *sa, SlackObject *obj, json_value *json,
 		json_value *old_message = json_get_prop(json, "previous_message");
 		/* this may consist only of added attachments, no changed text */
 		gboolean changed = g_strcmp0(json_get_prop_strptr(message, "text"), json_get_prop_strptr(old_message, "text"));
-		g_string_append(html, "<font color=\"#717274\"><i>[edit] ");
-		if (old_message && changed) {
-			g_string_append(html, "(Old message: ");
-			slack_json_to_html(html, sa, old_message, NULL);
-			g_string_append(html, ")<br>");
+		// No change means that this is a link update, which we want to suppress.
+		if (changed) {
+			g_string_append(html, "<font color=\"#717274\"><i>[edit] ");
+			if (old_message) {
+				g_string_append(html, "(Old message: ");
+				slack_json_to_html(html, sa, old_message, NULL);
+				g_string_append(html, ")<br>");
+			}
+			g_string_append(html, "</i></font>");
+			slack_json_to_html(html, sa, message, &flags);
```

Comment:
I think this needs to stay outside the conditional so we still get some message.

Reply:
Why? This defeats the purpose. The point is to reduce noise by not producing any message when the old and the new message are identical. This won't happen for edits, but it will happen when attachments are added which we strip away (the `expand_urls` option), leaving us with the same message as we started with.

Reply:
The problem was that even when `expand_urls` was enabled, it still didn't display the url expansion, because `slack_json_to_html` was never called on the new message. I just added an option check in there instead.

Reply:
Ah, doh! 🤦 How did I not see that? Thanks!